clients/python-wrapper/lakefs/branch.py

```diff
@@ -167,11 +169,27 @@ def delete_objects(self, object_paths: str | StoredObject | Iterable[str | Store
         """
         Delete objects from lakeFS
 
+        This method can be used to delete single/multiple objects from branch. It accepts both str and StoredObject
+        types as well as Iterables of these types.
+        Using this method is more performant than sequentially calling delete on objects as it saves the back and forth
+        from the server.
```

Comment:
- Underlaying code uses delete_objects directly and do not chunk the iterables into the server limit. An example where more than 1k object listing will break the delete. Also in terms on memory usage it is not optimal.

We can update the code of delete to handle the limit case and use adapter / lambda to get iterable that will return a path from listing.

Also not part of this PR, we can call delete_objects API on request to delete a single object - the usage pattern that we see in our API also means helps us understand how the system is used.

Reply:
Added max_amount to the query. I think the example code should be simple and not over-complex and should not provide a complete solution

Reply:
It stand as simple example, but my concern is that usually people learn and use the example for their usage.

Would prefer the library code to handle this case - passing array of 2k items without breaking

Reply:
The limitation is on the server side and might change - I don't think we want to have to maintain this on the SDK side every time we make changes on the server. Let the server handle it and return the error